lib/src/tree.rs

```diff
@@ -552,6 +552,15 @@ pub fn merge_trees(
     store.write_tree(dir, &new_tree)
 }
 
+/// Returns `Some(...)` if this is a directory or missing.
+fn maybe_tree_id(value: Option<&TreeValue>) -> Option<Option<&TreeId>> {
```

Comment:
Option<Option<_>> Strikes me as odd. I think I'd encode it as a separate enum, that is:

enum MaybeTree {
  Missing,
  Tree(TreeId),
}

or similar. WDYT?

Reply:
Thanks for reviewing! Yeah, I agree that the `Option<Option<_>>` was ugly. I was thinking when I wrote the code that it would be cleaner to return the ID for the empty tree there instead of `Some(None)`, but I thought it would be a bit wasteful to get the tree ID if it might not be used. It's just a reference, though, so I don't know what I was thinking (maybe I was actually thinking about returning the whole tree?). I've cleaned it up accordingly now.